frontend/src/app/shared/functions/cypress.ts

```diff
@@ -0,0 +1,24 @@
+import type { Store } from '@ngrx/store';
+import type { ExerciseAction, Immutable } from 'digital-fuesim-manv-shared';
+import { environment } from 'src/environments/environment';
+
+export interface CypressTestingValues {
+    store: Store;
+    proposedActions: Immutable<ExerciseAction>[];
+    performedAcctions: Immutable<ExerciseAction>[];
+    backendBaseUrl: string;
+}
+
+export const isBeingTestedByCypress = () =>
+    'Cypress' in window && !environment.production;
+
+export const setupCypressTestingValues = (
+    apply: (values: Partial<CypressTestingValues>) => void
+) => {
+    const anyWindow = window as any;
+    if (isBeingTestedByCypress()) {
+        if (!anyWindow.cypressTestingValues)
+            anyWindow.cypressTestingValues = {};
+        apply(anyWindow.cypressTestingValues);
+    }
+};
```

Comment:
Instead of an `apply` function, directly pass an `Partial<CypressTestingValues>.
Use `defaults()` from lodash to add it to the window property via `window.cypressTestingValues = defaults(values, window.cypressTestingValues)`.

If the function currently has side effects, one should do it separately from setting the values with an additional check of `isBeingTestedByCypress()`.

This should make this API easier to understand, as it is in this codebase usually discouraged to mutate function arguments.

Reply:
I've changed it in 8c05375cce5b6456dc1c970b594b7c028fd5dd63, do you think this is better now?